src/redux/services/templates.ts

```diff
@@ -128,24 +129,24 @@ export const createUnsavedResourcesFromVanillaTemplate = async (
       }
     }
   );
-  const createdResources: K8sResource[] = [];
+
+  let objects: any[] = [];
```

Comment:
there is a `PossibleResource` type in the `redux/services/manifest-utils` file, could you use that here?
maybe move it to the `models/appstate` first and then import it in both places.

also add something like this to be sure the properties exist:
```
obj &&
typeof obj.apiVersion === 'string' &&
typeof obj.kind === 'string' &&
typeof obj.metadata?.name === 'string'
```

You could probably extrat that into a type guard named `isPossibleResource` and have it in the `appstate` as well

Reply:
where do you think to use this check? Do we need it really?